enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,442 @@
+>this proposal may occur during the review process.
+> 4. We did *NOT* provide for multiple sets of entitlement credentials in 3.x.  There have been no official, RFE style  
+>registered requirements entitlements at a namespace level.  However, we've started to get (slightly) less official
+>feedback that customers want this in a more automated sense (vs. the current manual steps).  At a minimum, and is
+>noted below when alternatives are discussed, if namespaced entitlements are added with subsequent implementation
+>after global entitlements usability is improved, it needs to be seamless, and we minimally need to agree now
+>on what the order of precedence will be (with the take on that listed below).
+> 5. There has also been a question in the past needing to shard entitlements due to API rate limiting.  To date, it
+>has been stated that is not a concern for builds.  Periodic checkpoints here could be prudent.
+> 6. There is also the general question of how entitlements are propagated for our managed service offerings.  There are
+some attempts at capturing the particulars for this below, but it needs some vetting.
+> 7. With the CSI plugin based solution articulated below, there is some question as to how that is packaged, such as a)
+> a separate admission server (possibly aggregated apiserver) utilizing
+>libraries like https://github.com/openshift/generic-admission-server, or b) perhaps even a patch we carry on the k8s
+>apiserver (yikes!!)
+
+## Summary
+
+So in V3.x, all OpenShift Builds essentially had the customer RHEL entitlements mounted into the build pod automatically
+and without further intervention after install "for free" because:
+
+ - the ansible based installer required/forced providing your subscription information in order to install OCP
+ - for reference, see [these docs](https://docs.openshift.com/container-platform/3.11/install/host_preparation.html#host-registration) 
+ and [these roles](https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/rhel_subscribe/tasks/main.yml)
+ - next, Red Hats's docker daemon installed in 3.x had specific logic that auto-mounted the host's subscriptions 
+ into the container
+
+In V4.x, with 
+
+- the removal of the docker daemon, meaning no docker socket and no access to the host's content
+- and the total replacement of the openshift ansible based installer, so no gathering of entitlements during install
+- and the total change in the OCP subscription model between 3.x and 4.x:  you don't subscribe RHCOS nodes (in fact they
+do not have or ever will the entire set of subscription manager RPMs, and any as discussed in some of the alternatives
+below, at most we will only seed enough subscription manager artifacts in RHCOS so that things like turning auto mount
+on won't SEGFAULT).  In 3.x, the underlying linux was classic RHEL, and so hence the subscription manager was fully 
```

Comment:
This needs to be thoroughly documented, I had customers who logged into RHCOS and wanted to entitle the node, with either Satellite or the subscription *.pem files.